File: hot_restart/overpainting.m
% A matrix
% v vector of priorities
function [Ac,Ar] = overpainting(A,v)
[m,n] = size(A);
[i,j,s] = find(A);
%[i1,index1] = sort(i);
%j1 = j(index1); 
%s1 = s(index1); 
for z=m+n:-1:1;
    k = v(z);
    if k<=m
        %it's a row
      
        s(i==k) = 10;
    else
        %it's a column
        s(j==(k-m)) = 11;
        
    end
end
ir = i(s==10);
jr = j(s==10);

ic = i(s==11);
jc = j(s==11);

Ar = sparse(ir,jr,ones(size(ir)),m,n);
Ac = sparse(ic,jc,ones(size(ic)),m,n);

File: hot_restart/overpainting.c
#include <stdlib.h>
#include <stdio.h>
#include <memory.h>
#include <math.h>
#include <stdlib.h>
#include <time.h>
#include <Mondriaan.h>
#include "../utils.c"


struct twomatrices overpaint(struct sparsematrix* A, long* vec){
  int m = A->m;
  int n = A->n;

  struct sparsematrixplus matrixplus = reorder_col_incr(A);
  struct sparsematrix* B = &(matrixplus.matrix);
  long* BtoA = matrixplus.perm;

  int i,k;

  for(i=m+n-1;i>=0;i--){
    k = vec[i];
    if (k<m){
      /* it's a row*/
      update_rows(A,k,10.0);
    } else {
      /* it's a column */
      update_cols_link(B,A,k-m,11.0,BtoA);
    }
  }

  return split_matrix(A,10.0,11.0);
}
/*
struct twomatrices overpaint_col(struct sparsematrix* A, long* vec){
  int m = A->m;
  int n = A->n;
  int length = A->NrNzElts;

  struct sparsematrixplus matrixplus = reorder_row_incr(A);
  struct sparsematrix* B = &(matrixplus.matrix);
  long* BtoA = matrixplus.perm;
  int i,k;
 

  for(i=length-1;i>=0;i--){
    k = vec[i];
    if (k<m){
      // it's a row
      update_rows_link(B,A,k,10.0,BtoA);
    } else {
      // it's a column 
      update_cols(A,k-m,11.0);
    }
    printf("===%d===\n",length-i-1);
    print_matrix(*A);
  }


  return split_matrix(A,10.0,11.0);
}*/